webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/inventory/InventoryFileWatcher.scala

```diff
@@ -109,19 +109,27 @@ object Watchers {
         // When we call "stop" on the FileMonitor, it always throws a ClosedWatchServiceException.
         // It seems to be because the "run" in start continue to try to execute its
         // action on the stop watcher. We need to catch that.
-        override def start()(implicit executionContext: ExecutionContext) = {
-          this.watch(root, 0)
-          executionContext.execute(new Runnable {
-            override def run() = {
-              try {
-                blocking { Iterator.continually(service.take()).foreach(process) }
-              } catch {
+        // Execution context is not used here - binding to scala default global
+        @silent("private default argument.*is never used")
+        override def start()(implicit executionContext: ExecutionContext = scala.concurrent.ExecutionContext.global): Unit = {
```

Comment:
Have you tried removing the execution context ?

Reply:
it's in parent method signature, I can't. But I moved it to call site, it avoids half of them.

Reply:
Ha yes, we override better-files file monitor